src/history/hs_rec.c

```diff
@@ -668,20 +668,26 @@ __wt_hs_delete_key_from_ts(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint3
     wt_timestamp_t hs_ts;
     uint64_t hs_counter;
     uint32_t hs_btree_id;
-    bool hs_read_committed;
+    bool hs_read_all_flag, hs_read_committed;
 
     /*
      * If we will delete all the updates of the key from the history store, we should not reinsert
      * any update.
      */
     WT_ASSERT(session, ts > WT_TS_NONE || !reinsert);
 
+    hs_read_all_flag = F_ISSET(hs_cursor, WT_CURSTD_HS_READ_ALL);
```

Comment:
No need to set WT_CURSTD_HS_READ_COMMITTED flag if we set WT_CURSTD_HS_READ_ALL.

Reply:
Can you run a mongodb patch build?